File: NetworkingPractice/NetworkingPractice/ViewController.m
//
//  ViewController.m
//  NetworkingPractice
//
//  Created by Yao Li on 10/12/16.
//  Copyright © 2016 clouds. All rights reserved.
//

#import "ViewController.h"
#import "ServerAgent.h"
#import "Utility.h"
#import "UIImageView+AFNetworking.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.

//    [self getMovie:@"batman"];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)getMovie:(NSString *)searchTerm {
    [ServerAgent getMovie:searchTerm callback:^(NSArray *movies) {
        if (movies && ![movies isEqual:[NSNull null]]) {
//            NSLog(@"Movies: %@", movies);
            if (movies.count > 0) {
                NSDictionary *movieObj = movies[0];
                NSLog(@"%@, %@, %@, %@, %@", movieObj[@"Poster"], movieObj[@"Title"], movieObj[@"Type"], movieObj[@"Year"],
                        movieObj[@"imdbID"]);
                _titleLabel.text = movieObj[@"Title"];
                [_postImgView setImageWithURL:[NSURL URLWithString:movieObj[@"Poster"]] placeholderImage:nil];
//              _postImgView.image =  [UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL fileURLWithPath:<#(NSString *)path#>]]]

            }
//            for (int i = 0; i < movies.count; i++) {
////                NSLog(@"%d: %@", (int)i, results[i]);
//                NSDictionary *movieObj = movies[i];
//                NSLog(@"%@, %@, %@, %@, %@", movieObj[@"Poster"], movieObj[@"Title"], movieObj[@"Type"], movieObj[@"Year"],
//                        movieObj[@"imdbID"]);
//            }
        } else {
            [Utility blankAlertWithMessage:@"Error" message:@"Try again later" owner:self];
        }
    }];
}

- (IBAction)searchTapped:(id)sender {
    NSString *searchTerm = _searchField.text;
    NSLog(@"%@", searchTerm);
    searchTerm = [searchTerm stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLHostAllowedCharacterSet]];
    NSLog(@"after: %@", searchTerm);
    [self getMovie:searchTerm];
}
@end

File: HitList/HitList/ViewController.swift
//
//  ViewController.swift
//  HitList
//
//  Created by Yao Li on 11/27/16.
//  Copyright © 2016 clouds. All rights reserved.
//

import UIKit
import CoreData

class ViewController: UIViewController, UITableViewDataSource {
    @IBOutlet weak var tableView: UITableView!
    var people = [NSManagedObject]()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        title = "\"The List\""
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func saveName(name: String) {
        //1
        let appDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
        let managedContext = appDelegate.managedObjectContext

        //2
        let entity =  NSEntityDescription.entityForName("Person", inManagedObjectContext:managedContext)
        let person = NSManagedObject(entity: entity!, insertIntoManagedObjectContext: managedContext)

        //3
        person.setValue(name, forKey: "name")

        //4
        do {
            try managedContext.save()
            //5
            people.append(person)
        } catch let error as NSError  {
            print("Could not save \(error), \(error.userInfo)")
        }
    }

    @IBAction func addName(_ sender: Any) {
        let alert = UIAlertController(title: "New Name", message: "Add a new name", preferredStyle: .alert)

        let saveAction = UIAlertAction(title: "Save", style: .default, handler: { (action:UIAlertAction) -> Void in
                    let textField = alert.textFields!.first
                    self.saveName(name:textField!.text!)
                    self.tableView.reloadData()
                })

        let cancelAction = UIAlertAction(title: "Cancel", style: .default) { (action: UIAlertAction) -> Void in }

        alert.addTextField(configurationHandler: {(textField: UITextField) -> Void in })

        alert.addAction(saveAction)
        alert.addAction(cancelAction)

        present(alert, animated:true, completion:nil)
    }

    // MARK: UITableViewDataSource
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return people.count
    }

    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell")

        let person = people[indexPath.row]
        cell!.textLabel!.text = person.value(forKey: "name") as? String

        return cell!
    }
}

